### DISCUSSION TITLE
PHP Fatal error: Cannot redeclare with opcache.preload and Composer autoload between releases

### DISCUSSION THREAD

**Discussion by gremo** [deployphp/deployer#3591] [Help needed]
- Deployer version:  7.3.1
- Deployment OS: Centos 7

My project is Symfony based, and uses the following opcache configuration:

```ini
opcache.preload = /var/www/vhosts/myproject/httpdocs/current/config/preload.php
```

It works the first time I activate the configuration. Next time I'll deply a new version, you got errors in PHP logs:

> AH01071: Got error 'PHP message: PHP Fatal error: Cannot redeclare composerRequiree7dcbbd7b36b37e14629ddf6c0ff22ac() (previously declared in /var/www/vhosts/myproject/httpdocs/releases/1/vendor/composer/autoload_real.php:73) in /var/www/vhosts/myproject/httpdocs/releases/2/vendor/composer/autoload_real.php on line 73'

As you can see the **release** 1 "conflict" with **release 2** because the same function is redeclared and still in memory (I think). 

Adding `after('deploy:symlink', 'cachetool:clear:opcache')` doesn't help to solve the issue.

What I'missing?

**Comment by antonmedv** [comment]
Try this one: [https://ï.at/avoid-php-fpm-reloading](https://xn--gda.at/avoid-php-fpm-reloading)

**Reply by gremo** [reply]
Before changing the server configuration, how that would solve the double-loading?

**Reply by tripany** [reply]
Did you ever solve this?

**Reply by gremo** [reply]
Yeah with the help of: https://deployer.org/docs/7.x/contrib/cachetool

**Reply by tripany** [reply]
Thanks,

I've tried `after('deploy:symlink', 'cachetool:clear:opcache')`, and also `cachetool:clear:apcu`, as well as `cachetool:clear:stat` without luck. 

I've also made sure to set and verify the nginx realpath config change.

Did you happen to configure anything else?

**Reply by gremo** [reply]
@tripany  this is the relevant part of my deployer file:

```php
// Configuration
set('cachetool_url', 'https://github.com/gordalina/cachetool/releases/download/7.1.0/cachetool.phar');
set('cachetool_args', '--web --web-path=./public --web-url=http://{{hostname}}');

// Tasks
after('deploy:symlink', 'cachetool:clear:opcache');
```

If you still have problem, you could try updating the framework, or start a new minimal project with Symfony 6 and test this configuration.
